### ISSUE TITLE
ACID(42) bodeplots contains no data (HG1/R2014a only)

### ISSUE THREAD

**Issue by PeterPablo** [matlab2tikz/matlab2tikz#638]
Only running ACID(42) and manually invoking matlab2tikz is almost fine. @egeerardyn, could you please check, since you seem to have the toolbox available.

```latex
% This file was created by matlab2tikz.
%
\documentclass[tikz]{standalone}
\usepackage[T1]{fontenc}
\usepackage[utf8]{inputenc}
\usepackage{pgfplots}
\usepackage{grffile}
\pgfplotsset{compat=newest}
\usetikzlibrary{plotmarks}
\usepgfplotslibrary{patchplots}
\usepackage{amsmath}

\newlength\figureHeight \setlength{\figureHeight}{6cm}
\newlength\figureWidth \setlength{\figureWidth}{10cm}
\begin{document}
\begin{tikzpicture}

\begin{axis}[%
width=0.982\figureWidth,
height=0.51\figureHeight,
at={(0\figureWidth,0.49\figureHeight)},
scale only axis,
unbounded coords=jump,
separate axis lines,
every outer x axis line/.append style={white!40!black},
every x tick label/.append style={font=\color{white!40!black}},
xmode=log,
xmin=   0,
xmax=   1,
xtick={ 0.1,   1},
xticklabels={\empty},
xminorticks=true,
xmajorgrids,
xminorgrids,
every outer y axis line/.append style={white!40!black},
every y tick label/.append style={font=\color{white!40!black}},
ymin=   0,
ymax=   1,
ylabel={Magnitude (dB)},
ymajorgrids,
axis background/.style={fill=white}
]
\addplot [color=blue,solid,line width=1.5pt,forget plot]
  table[row sep=crcr]{%
 nan	 nan\\
};
\addplot [color=black,solid,line width=1.5pt,forget plot]
  table[row sep=crcr]{%
 nan	 nan\\
};
\addplot [color=black!35!green,solid,line width=1.5pt,forget plot]
  table[row sep=crcr]{%
 nan	 nan\\
};
\addplot [color=black,solid,line width=1.5pt,forget plot]
  table[row sep=crcr]{%
 nan	 nan\\
};
\end{axis}

\begin{axis}[%
width=0.982\figureWidth,
height=0.453\figureHeight,
at={(0\figureWidth,0\figureHeight)},
scale only axis,
unbounded coords=jump,
separate axis lines,
every outer x axis line/.append style={white!40!black},
every x tick label/.append style={font=\color{white!40!black}},
xmode=log,
xmin=   0,
xmax=   1,
xminorticks=true,
xmajorgrids,
xminorgrids,
every outer y axis line/.append style={white!40!black},
every y tick label/.append style={font=\color{white!40!black}},
ymin=   0,
ymax=   1,
ylabel={Phase (deg)},
ymajorgrids,
axis background/.style={fill=white},
legend style={at={(0.03,0.03)},anchor=south west,legend cell align=left,align=left,draw=black}
]
\addplot [color=blue,solid,line width=1.5pt]
  table[row sep=crcr]{%
 nan	 nan\\
};
\addlegendentry{};

\addplot [color=black,solid,line width=1.5pt]
  table[row sep=crcr]{%
 nan	 nan\\
};
\addlegendentry{};

\addplot [color=black!35!green,solid,line width=1.5pt]
  table[row sep=crcr]{%
 nan	 nan\\
};
\addlegendentry{};

\addplot [color=black,solid,line width=1.5pt]
  table[row sep=crcr]{%
 nan	 nan\\
};
\addlegendentry{};

\end{axis}
\end{tikzpicture}%
\end{document}
```

**Comment by egeerardyn** [comment]
I can confirm this behavior.
Calling
```matlab
figure, ACID(42), matlab2tikz('data/converted/test42-converted.tex')
```
seems to work fine, but both `testHeadless(42)` and `testGraphical(42)` don't.

After some searching, this seems to be caused by invisible figures. Those plots are initialized with `NaN` data and the actual data are only loaded by means of a callback. For invisible figures, it seems that this callback is never executed.

**Comment by PeterPablo** [comment]
Nice findings! Do we want to support this? I guess the function handle is present somewhere in the figure -- calling it might be dangerous though, if this would change an existing plot.
I see you started working on this -- looking forward to your proposed solution.

**Comment by egeerardyn** [comment]
Well, this is just an annoyance for us devs. It's better to get it out of the way to avoid confusion in the future. I don't plan to call the callback by hand, but rather as I show above. I've already tested that on my install and it seems to work.

**Comment by PeterPablo** [comment]
fixed by aa00b38
